fluidsynth/src/sfloader/fluid_defsfont.h

```diff
@@ -435,7 +435,7 @@ struct _fluid_defpreset_t
 {
   fluid_defpreset_t* next;
   fluid_defsfont_t* sfont;                  /* the soundfont this preset belongs to */
-  char name[21];                        /* the name of the preset */
+  char name[22];                        /* the name of the preset */
```

Comment:
> the compiler assumes that the arguments bank and prenum can become up to 5 bytes in size, each.

4 bytes, but 5 decimal digits.

> 7.2 The PHDR Sub-chunk
> ...
> If the value in [wBank and wPreset] is not a valid MIDI value of zero through 127, or 128 for wBank, the preset cannot be played but should be maintained.

Which means they can have any value of 0 to 65536 and must be maintained (i.e. loaded; and from the code it seems fluidsynth would really load such presets...) although they become unusable because a midi prog change only ranges from 0 to 127. However for the bank selection having CC0 and CC32 we should be able to address up to `127*128+127=16383` banks in mma mode.

Anyway, internally I would prefer to have uniquely named preset, i.e. not doing `sfpreset->prenum % 128`.